src/reducers/existingThoughtChange.js

```diff
@@ -43,18 +45,21 @@ export default (state, { oldValue, newValue, context, showContexts, thoughtsRank
     ? contextOf(contextOf(thoughtsRanked)).concat({ value: oldValue, rank: headRank(contextOf(thoughtsRanked)) }).concat(head(thoughtsRanked))
     : contextOf(thoughtsRanked).concat({ value: oldValue, rank })
 
+  /** Find exact thought from thoughtIndex. */
+  const exactThought = () => thoughtOld.contexts.find(thought => equalArrays(thought.context, context) && thought.rank === rank)
+  const id = headId(thoughtsRanked) || exactThought().id
```

Comment:
Since `exactThought` is always called, you can use a normal variable rather than a function:

```js
const exactThought = thoughtOld.contexts.find(thought => equalArrays(thought.context, context) && thought.rank === rank)
```